### PR TITLE
Fix autodiscovery vs discovery on LLDP/xDP links

### PR DESCRIPTION
**Pull request by kedare**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by kedare** [comment]
This PR make LLDP discovery work as expected, discovering the links as advertised by LLDP/xDP, but trying to discover/add the neighbour only if the specific configuration parameter is set properly.

Related to https://community.librenms.org/t/how-to-disable-lldp-autodiscovery-but-not-lldp-discovery/10424/13

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/how-to-disable-lldp-autodiscovery-but-not-lldp-discovery/10424/18

**Comment by kedare** [comment]
Tested on my setup with LLDP links, working as intended (I don't have the others discovery protocols so I cannot test them)

**Comment by kedare** [comment]
Hmm looks like I have some CDP running, I still see my logs populated by a `CDP discovery of Switch failed - Check name lookup` but I don't know why

**Comment by laf** [comment]
@kedare You sure you don't have the config option enabled?

Also, missing one of the checks: https://github.com/librenms/librenms/blob/7d9829f1d41174d0b88ec7736317f7face2955fe/includes/discovery/discovery-protocols.inc.php#L338

File: doc/Extensions/Auto-Discovery.md [hunk 1]
```diff
@@ -107,6 +107,9 @@ Enabled by default.
 
 This includes FDP, CDP and LLDP support based on the device type.
 
+The LLDP/xDP links with neighbours will always be discovered as soon as the discovery module is enabled.
+However, LibreNMS will only try to add the new devices discovered with LLDP/xDP only if `autodiscovery.xdp` is set to true.
```
Comment (laf) [inline doc/Extensions/Auto-Discovery.md]: Should be if `$config['autodiscovery']['xdp'] = true;` is set.

**Comment by kedare** [comment]
For the OSPF thing I don't think this need to be updated, as OSPF is polled correctly from the ospf discovery module and this code is only used to try to add the neighbors to the monitoring (so working as expected).

File: doc/Extensions/Auto-Discovery.md (hunk 1 above)
Comment (kedare) [reply doc/Extensions/Auto-Discovery.md]: Fixed

**Comment by kedare** [comment]
Looks like CI has timed out ?

**Comment by louis-oui** [comment]
> Looks like CI has timed out ?

Do a dummy git push like pushing something useless

**Comment by kedare** [comment]
Done

**Comment by laf** [comment]
Need to provide notification to users on this so don't emerge just yet.

**Comment by kedare** [comment]
Done and merged https://github.com/librenms/librenms.github.io/pull/126

**Comment by zombah** [comment]
Tested PR with release 1.61 and autodiscovery lldp disabled.
Seems to work as expected, no more attempts in device events to add any new lldp neighbor.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-62-release-changelog-march-2020/11423/1

**Pull request merged**